chain/arweave/src/chain.rs

```diff
@@ -44,20 +44,20 @@ impl std::fmt::Debug for Chain {
     }
 }
 
-impl Chain {
-    pub fn new(
+impl BuildableBlockchain for Chain {
+    fn build(
         logger_factory: LoggerFactory,
-        name: String,
+        chain_id: String,
```

Comment:
This is the network_name, there is also a chain_id whihc is something internal to each chain, I think we should keep the previous naming